bin/ocaml/top.ml

```diff
@@ -100,6 +100,11 @@ module Module = struct
     let ctx = Super_context.context sctx in
     let src = Path.Build.append_source ctx.build_dir mod_ in
     let dir = Path.Build.parent_exn src in
+    let filename = Path.Build.basename src in
+    if Sys.file_exists filename then ()
+    else User_error.raise [ Pp.text "file doesn't exist" ] ;
+    if Filename.extension filename  = ".ml" then ()
+    else User_error.raise [ Pp.text "file should have '.ml' extension" ] ;
     let open Memo.O in
     let module_name =
       let name = src |> Path.Build.basename |> Filename.chop_extension in
```

Comment:
This fix isn't quite right because we allow other dialects apart from OCaml. ReasonML is one example. The file existence checks are also done later so they aren't needed.

I think we should just reject files without an extension and be done with it. Could you update the PR to do that?

Reply:
Done